command/snapshot/save/snapshot_save.go

```diff
@@ -23,17 +26,26 @@ func New(ui cli.Ui) *cmd {
 }
 
 type cmd struct {
-	UI    cli.Ui
-	flags *flag.FlagSet
-	http  *flags.HTTPFlags
-	help  string
+	UI                 cli.Ui
+	flags              *flag.FlagSet
+	http               *flags.HTTPFlags
+	help               string
+	appendFileNameFlag flags.StringValue
+}
+
+func (c *cmd) getAppendFileNameFlag() *flag.FlagSet {
+	fs := flag.NewFlagSet("", flag.ContinueOnError)
+	fs.Var(&c.appendFileNameFlag, "append-filename", "Append filename flag takes two possible values. "+
+		"1. version, 2. dc. It appends consul version and datacenter to filename given in command")
```

Comment:
Should we add validation to check if other fields or present?

Reply:
I prefer giving results based on given flags.